src/winetricks

```diff
@@ -15052,6 +15052,105 @@ load_steam()
+  ## TODO: file a winebug
+
+  # non-legacy download
+  #w_download 'https://origin-a.akamaihd.net/Origin-Client-Download/origin/live/OriginThinSetup.exe'
+
+  # legacy download
+  w_download 'https://origin-a.akamaihd.net/Origin-Client-Download/origin/legacy/OriginThinSetup.exe'
+
+  # NOTICE: Both legacy and non-legacy are using dx11 and dxgi libs
+
+  # Logic for DXVK
+  if [ -n "$WINETRICKS_ORIGIN_USE_DXVK" ]; then
+    w_call 'dxvk'
+
+    # Installer fails without overrides to OriginClientService assuming missing function in WINE
+    # TODO: File a winebug
+    w_override_app_dlls 'OriginClientService.exe' 'disabled' 'd3d10'
+    w_override_app_dlls 'OriginClientService.exe' 'disabled' 'd3d10_1'
+    w_override_app_dlls 'OriginClientService.exe' 'disabled' 'd3d10core'
+    w_override_app_dlls 'OriginClientService.exe' 'disabled' 'd3d11'
+    w_override_app_dlls 'OriginClientService.exe' 'disabled' 'dxgi'
+
+  elif [ -z "$WINETRICKS_ORIGIN_USE_DXVK" ]; then
+    w_info "DXVK was disabled for origin client, to enable set WINETRICKS_ORIGIN_USE_DXVK variable to non-zero"
+
+    # Disable dxvk libraries if variable above is not set
+    w_override_app_dlls 'Origin.exe' 'disabled' 'd3d10'
+    w_override_app_dlls 'Origin.exe' 'disabled' 'd3d10_1'
+    w_override_app_dlls 'Origin.exe' 'disabled' 'd3d10core'
+    w_override_app_dlls 'Origin.exe' 'disabled' 'd3d11'
+    w_override_app_dlls 'Origin.exe' 'disabled' 'dxgi'
+
+    w_override_app_dlls 'IGOProxy.exe' 'disabled' 'd3d10'
+    w_override_app_dlls 'IGOProxy.exe' 'disabled' 'd3d10_1'
+    w_override_app_dlls 'IGOProxy.exe' 'disabled' 'd3d10core'
+    w_override_app_dlls 'IGOProxy.exe' 'disabled' 'd3d11'
+    w_override_app_dlls 'IGOProxy.exe' 'disabled' 'dxgi'
+
+    w_override_app_dlls 'IGOProxy64.exe' 'disabled' 'd3d10'
+    w_override_app_dlls 'IGOProxy64.exe' 'disabled' 'd3d10_1'
+    w_override_app_dlls 'IGOProxy64.exe' 'disabled' 'd3d10core'
+    w_override_app_dlls 'IGOProxy64.exe' 'disabled' 'd3d11'
+    w_override_app_dlls 'IGOProxy64.exe' 'disabled' 'dxgi'
+
+  fi
+
+  # Set environment on winxp for required apps
+  ## Using winvista is mandatory winetricks/1318#issuecomment-531562386
+  w_set_app_winver 'Origin.exe' 'winvista'
+  w_set_app_winver 'OriginClientService.exe' 'winxp'
+  w_set_app_winver 'Setup.exe' 'winvista'
+  w_set_app_winver 'UpdateTool.exe' 'winvista'
+  w_set_app_winver 'OriginUninstall.exe' 'winvista'
+  w_set_app_winver 'OriginCrashReporter.exe' 'winvista'
+  w_set_app_winver 'OriginClient.exe' 'winvista'
+  w_set_app_winver 'IGOProxy.exe' 'winvista'
+  w_set_app_winver 'IGOProxy64.exe' 'winvista'
+  w_set_app_winver 'OriginFR.exe' 'winvista'
+  w_set_app_winver 'GetGameToken32.exe' 'winvista'
+  w_set_app_winver 'GetGameToken64.exe' 'winvista'
+  w_set_app_winver 'EAProxyInstaller.exe' 'winvista'
+
+  # winxp is required for the installation
+  w_set_winver 'winxp'
+
+  w_warn "You should *uncheck* sharing hardware info"
```

Comment:
I'd suggest making a quiet version that sets this with ahk, and warns if a non quiet install.